zerver/tests/test_topic_visibility_policies.py

```diff
@@ -223,3 +228,49 @@ def test_muted_topic_remove_invalid(self) -> None:
         data = {"stream": stream.name, "stream_id": stream.id, "topic": "Verona3", "op": "remove"}
         result = self.api_patch(user, url, data)
         self.assert_json_error(result, "Please choose one: 'stream' or 'stream_id'.")
+
+
+class UnmutedVisibilityTests(ZulipTestCase):
+    def test_user_ids_unmuting_topic(self) -> None:
+        hamlet = self.example_user("hamlet")
+        cordelia = self.example_user("cordelia")
+        realm = hamlet.realm
+        stream = get_stream("Verona", realm)
+        recipient = stream.recipient
+        topic_name = "teST topic"
+
+        stream_topic_target = StreamTopicTarget(
+            stream_id=stream.id,
+            topic_name=topic_name,
+        )
+
+        user_ids = stream_topic_target.user_ids_with_visibility_policy(UserTopic.UNMUTED)
+        self.assertEqual(user_ids, set())
+
+        def set_topic_visibility_for_user(user: UserProfile, visibility_policy: int) -> None:
+            assert recipient is not None
+            add_topic_visibility_policy(
+                user_profile=user,
+                stream_id=stream.id,
+                recipient_id=recipient.id,
+                topic_name="test TOPIC",
+                visibility_policy=visibility_policy,
+                date_added=timezone_now(),
+            )
+
+        set_topic_visibility_for_user(hamlet, UserTopic.UNMUTED)
+        set_topic_visibility_for_user(cordelia, UserTopic.MUTED)
+        user_ids = stream_topic_target.user_ids_with_visibility_policy(UserTopic.UNMUTED)
+        self.assertEqual(user_ids, {hamlet.id})
+        hamlet_date_unmuted = UserTopic.objects.filter(
+            user_profile=hamlet, visibility_policy=UserTopic.UNMUTED
+        )[0].last_updated
+        self.assertTrue(timezone_now() - hamlet_date_unmuted <= timedelta(seconds=100))
```

Comment:
It's best to never use real time in tests. The 100s delay you have here should be fine, but a very large portion of flaky tests result from using real time. Just make up a fixed value for `date_added` to use inside `set_topic_visibility_for_user` and check that the value is as expected.

Reply:
**To Do**

Reply:
I have added a [prep commit](https://github.com/zulip/zulip/pull/24236/commits/17be5c18ebcfaf5afcc8acc6ee4cad2a9a4b6927) to replace the existing pattern of using real time. (in the existing tests in `test_muting_topic.py`)

Newer tests follow the updated pattern of using fixed value for `date_added`.